static/app/views/alerts/incidentRedirect.tsx

```diff
@@ -17,16 +17,18 @@ type Props = {
 /**
  * Reirects from an incident to the incident's metric alert details page
  */
-function IncidentRedirect({organization, params}: Props) {
+function IncidentRedirect({organization, params, location}: Props) {
   const api = useApi();
   const [hasError, setHasError] = useState(false);
+  const referrer = location.query?.referrer;
 
   const track = useCallback(() => {
     trackAdvancedAnalyticsEvent('alert_details.viewed', {
       organization,
       alert_id: parseInt(params.alertId, 10),
+      referrer: referrer || '',
```

Comment:
@Neo-Zhixing I think we automatically pass this in

Reply:
you can usually parse referrer from the url that is captured, depends how you want to query things

Reply:
@scefali updated